### ISSUE TITLE
Implement service monitoring support for java spring boot 2 and define timeoutSeconds for readinessProbe

### ISSUE THREAD

**Issue by Kamran64** [appsody/stacks#441]
Following https://github.com/appsody/appsody-operator/pull/125, the Appsody operator supports creating service monitor if there is a `spec.monitoring` entry in the `app-deploy.yaml` file. The only requirement to utilise this is a label field, like so:

```
monitoring:
    labels:
      k8s-app: APPSODY_PROJECT_NAME
```
We introduced this functionality for Python Flask, Node.js-Express, and Node.js-functions but it would be nice to see this for Java-Spring-Boot2 also.

Currently, we also see a Knative error if a stack uses the `readinessProbe` and `timeoutSeconds` is not defined.

**Describe the solution you'd like**
Add a new field (`monitoring`) to the `app-deploy` file underneath `spec` written as mentioned above.

Also add a new field, timeoutSeconds under the readinessProbe and assign it an appropriate value e.g. 1.

**Additional context**
Example PR to implement this into a stack: https://github.com/appsody/stacks/pull/419/files

**Note**: For stacks which have an authenticated endpoint, a `basicAuth` field is required, see: https://github.com/appsody/appsody-operator/pull/125

**Comment by Kamran64** [comment]
@BarDweller @ebullient 

Would you guys be happy to tackle this one and https://github.com/appsody/stacks/issues/442? I've linked an example implementation in the issue if you want to take a look.

**Comment by Kamran64** [comment]
Closing this issue. This work has been resolved in #491 and #490
